shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -204,16 +204,17 @@ - (void)setSystemChromeSystemUIOverlayStyle:(NSDictionary*)message {
 
 - (void)popSystemNavigator:(BOOL)isAnimated {
   // Apple's human user guidelines say not to terminate iOS applications. However, if the
-  // root view of the app is a navigation controller, it is instructed to back up a level
-  // in the navigation hierarchy.
+  // FlutterViewController or one of its ancestors is a child of a navigation controller,
+  // it is instructed to back up a level in the navigation hierarchy.
   // It's also possible in an Add2App scenario that the FlutterViewController was presented
   // outside the context of a UINavigationController, and still wants to be popped.
-  UIViewController* viewController = [UIApplication sharedApplication].keyWindow.rootViewController;
-  if ([viewController isKindOfClass:[UINavigationController class]]) {
-    [((UINavigationController*)viewController) popViewControllerAnimated:isAnimated];
+  auto engineViewController = static_cast<UIViewController*>([_engine.get() viewController]);
```

Comment:
The cast is unnecessary.

Reply:
I tried to change as little as possible, and as the cast was already being used, I kept it as it was.
However, the `viewController` property from `_engine.get` is `NSViewController` and we need it as `UIViewController`.

Reply:
Are you looking at FlutterViewController.h for [desktop](https://github.com/flutter/engine/blob/61ec840edd01e0c7e2be505f4f5d7fefb3c448c3/shell/platform/darwin/macos/framework/Headers/FlutterViewController.h#L29) vs [iOS](https://github.com/flutter/engine/blob/61ec840edd01e0c7e2be505f4f5d7fefb3c448c3/shell/platform/darwin/ios/framework/Headers/FlutterViewController.h#L51)? Desktop is where `NSViewController` is being using.